src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTask.java

```diff
@@ -249,4 +275,59 @@ private static String recordString(SinkRecord record) {
         record.kafkaOffset()
     );
   }
+
+  @Override
+  public void close(Collection<TopicPartition> partitions) {
+    offsetTracker.closePartitions(partitions);
+  }
+
+  // Visible for testing
+  static class PartitionPauser {
+
+    // Kafka consumer poll timeout to set when partitions are paused, to avoid waiting for a long
+    // time (default poll timeout) to resume it.
+    private static final long PAUSE_POLL_TIMEOUT_MS = 100;
+
+    private final SinkTaskContext context;
+    private final BooleanSupplier pauseCondition;
+    private final BooleanSupplier resumeCondition;
+    private boolean partitionsPaused;
+
+    public PartitionPauser(SinkTaskContext context,
+                           BooleanSupplier pauseCondition,
+                           BooleanSupplier resumeCondition) {
+      this.context = context;
+      this.pauseCondition = pauseCondition;
+      this.resumeCondition = resumeCondition;
+    }
+
+    /**
+     * Resume partitions if they are paused and resume condition is met.
+     * Has to be run in the task thread.
+     */
+    void maybeResumePartitions() {
+      if (partitionsPaused) {
+        if (resumeCondition.getAsBoolean()) {
+          log.debug("Resuming all partitions");
+          context.resume(context.assignment().toArray(new TopicPartition[0]));
+          partitionsPaused = false;
+        } else {
+          context.timeout(PAUSE_POLL_TIMEOUT_MS);
+        }
+      }
+    }
+
+    /**
+     * Pause partitions if they are paused and resume condition is met.
```

Comment:
```suggestion
     * Pause partitions if they are not paused and pause condition is met.
```

Reply:
Good catch, we don't have tests for javadoc :)